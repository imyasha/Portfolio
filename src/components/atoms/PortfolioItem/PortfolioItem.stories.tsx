import { PortfolioItem, PortfolioItemProps } from './PortfolioItem';

export default {
	title: 'atoms/PortfolioItem',
	component: PortfolioItem,
};

export const PortfolioItemStory = (args: PortfolioItemProps) => (
	<PortfolioItem {...args} />
);

PortfolioItemStory.storyName = 'PortfolioItem';
PortfolioItemStory.args = {
	item: {
		slug: 'amvi',
		title: 'Amamos Vivir Cada DÃ­a',
		link: 'https://www.amvi.org.mx/',
		type: 'Web Development',
		technologies: [
			'Gatsby',
			'TypeScript',
			'Styled-Components',
			'REST APIs',
			'GraphQL',
		],
	},
};
