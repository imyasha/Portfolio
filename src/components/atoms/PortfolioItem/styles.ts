import styled from 'styled-components';

const StyledPortfolioItem = styled.article`
	margin-bottom: 3rem;
	flex-direction: column;

	a {
		color: ${({ theme }) => theme.headline} !important;
		text-decoration: none;
	}

	h1 {
		margin: 0;
	}

	p {
		margin: 0;
	}

	.type {
		margin-bottom: 1rem;
		font-weight: bold;
	}

	@media screen and (min-width: 768px) {
		display: flex;
		flex-direction: row-reverse;
		align-items: center;

		&:nth-of-type(odd) {
			flex-direction: row;

			.details {
				padding-left: 1rem;
			}
		}

		&:nth-of-type(even) {
			.details {
				padding-right: 1rem;
			}
		}

		& > * {
			width: 50%;
		}
	}
`;

export { StyledPortfolioItem };
