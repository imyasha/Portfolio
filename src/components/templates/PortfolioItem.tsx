import React from 'react';
import styled from 'styled-components';
import { graphql, Link } from 'gatsby';
import Img from 'gatsby-image';
import { documentToReactComponents } from '@contentful/rich-text-react-renderer';

import SEO from 'Molecules/Seo';
import Container from 'Atoms/Container';
import Button from 'Atoms/Button';

interface IProps {
  data: {
    contentfulItem: {
      title: string;
      description: string;
      featured_image: {
        fluid: any;
      };
      type: string;
      content: {
        json: any;
      };
      technologies: string[];
      url: string;
    };
  };
  pageContext: {
    previous: {
      slug: string;
      title: string;
    } | null;
    next: {
      slug: string;
      title: string;
    } | null;
  };
}

const PortfolioItem = ({ data, pageContext }: IProps) => {
  const {
    title,
    description,
    featured_image: image,
    type,
    content: { json: content },
    technologies,
    url,
  } = data.contentfulItem;
  const { previous, next } = pageContext;

  return (
    <>
      <SEO title={title} description={description} />
      <Image fluid={image.fluid} alt={title} style={{ maxHeight: '400px' }} />
      <Container>
        <h2 data-testid="projectTitle">{title}</h2>
        <h3 data-testid="projectType">{type}</h3>
        <div data-testid="projectBody">
          {documentToReactComponents(content)}
        </div>
        <p data-testid="projectTechnologies">
          <strong>Technologies used:</strong> {technologies.join(', ')}
        </p>
        <Button
          href={url}
          target="_blank"
          rel="noopener noreferrer"
          data-testid="projectLink"
        >
          Visit project
        </Button>
        <Pagination>
          {previous && (
            <li data-testid="previous">
              <Link to={`/portfolio/${previous.slug}/`} rel="prev">
                &laquo; {previous.title}
              </Link>
            </li>
          )}
          {next && (
            <li data-testid="next">
              <Link to={`/portfolio/${next.slug}/`} rel="next">
                {next.title} &raquo;
              </Link>
            </li>
          )}
        </Pagination>
      </Container>
    </>
  );
};

export default PortfolioItem;

const Image = styled(Img)`
  margin-top: 2rem;
`;

const Pagination = styled.ul`
  list-style: none;
  margin: 3rem 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  li {
    a {
      color: #ccc;
      text-decoration: none;
      transition: 0.2s;
      &:hover {
        color: #999;
      }
    }
  }
`;

export const pageQuery = graphql`
  query PortfolioItemBySlug($slug: String!) {
    site {
      siteMetadata {
        title
        author
      }
    }
    contentfulItem(slug: { eq: $slug }) {
      title
      client
      type
      technologies
      url
      description
      content {
        json
      }
      featured_image {
        fluid(maxWidth: 1920, quality: 70) {
          ...GatsbyContentfulFluid
        }
      }
    }
  }
`;
