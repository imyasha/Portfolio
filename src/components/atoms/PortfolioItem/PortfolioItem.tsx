import { FC } from 'react';
import Image from 'next/image';
import { IPortfolioItem } from '@Types';
import Link from 'next/link';
import { shimmer } from 'Utils/shimmer';
import { StyledPortfolioItem } from './styles';

export interface PortfolioItemProps {
	item: IPortfolioItem;
}

const PortfolioItem: FC<PortfolioItemProps> = ({ item }) => {
	const { slug, title, type, technologies } = item;

	return (
		<StyledPortfolioItem>
			<Link href={`/portfolio/${slug}`}>
				<a>
					<Image
						src={`/portfolio/${slug}.jpg`}
						width={1920}
						height={1080}
						objectFit="cover"
						alt={title}
						placeholder="blur"
						blurDataURL={shimmer(1920, 1080)}
					/>
				</a>
			</Link>
			<div className="details">
				<Link href={`/portfolio/${slug}`}>
					<a>
						<h1>{title}</h1>
						<p className="type">{type}</p>
						<p>
							<strong>Technologies used:</strong>{' '}
							<em>{technologies.join(', ')}</em>
						</p>
					</a>
				</Link>
			</div>
		</StyledPortfolioItem>
	);
};

export { PortfolioItem };
