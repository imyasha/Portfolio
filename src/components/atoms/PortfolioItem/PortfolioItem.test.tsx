import { PortfolioItem } from '../PortfolioItem';
import { render } from '@testing-library/react';

describe('PortfolioItem', () => {
	const mockItem = {
		slug: 'amvi',
		title: 'Amamos Vivir Cada DÃ­a',
		link: 'https://www.amvi.org.mx/',
		type: 'Web Development',
		technologies: [
			'Gatsby',
			'TypeScript',
			'Styled-Components',
			'REST APIs',
			'GraphQL',
		],
		content: 'Some text content',
	};

	it('renders correctly', () => {
		const { container } = render(<PortfolioItem item={mockItem} />);
		expect(container).toMatchSnapshot();
	});
});
