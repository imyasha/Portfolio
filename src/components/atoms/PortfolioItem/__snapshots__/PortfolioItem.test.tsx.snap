// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioItem renders correctly 1`] = `
<div>
  <article
    class="styles__StyledPortfolioItem-n6dl7r-0 dBellb"
  >
    <a
      href="/portfolio/amvi"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="Amamos Vivir Cada Día"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </div>
    </a>
    <div
      class="details"
    >
      <a
        href="/portfolio/amvi"
      >
        <h1>
          Amamos Vivir Cada Día
        </h1>
        <p
          class="type"
        >
          Web Development
        </p>
        <p>
          <strong>
            Technologies used:
          </strong>
           
          <em>
            Gatsby, TypeScript, Styled-Components, REST APIs, GraphQL
          </em>
        </p>
      </a>
    </div>
  </article>
</div>
`;
